py/selenium/webdriver/firefox/service.py

```diff
@@ -43,12 +44,26 @@ def __init__(self, executable_path, port=0, service_args=None,
         """
         log_file = open(log_path, "a+") if log_path is not None and log_path != "" else None
 
+        if port != 0:
+            import warnings
```

Comment:
this structure for deprecation warning was taken from `firefox_profile.py`

if it is not correct, I am happy to change it!